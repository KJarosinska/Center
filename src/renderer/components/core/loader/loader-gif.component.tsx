import * as React from "react"
import FunctionComponent from "Renderer/types/function-component.interface"
import { ImgHTMLAttributes } from "react"
import Image from "Renderer/components/core/image/image.component"
import Gif from "Renderer/images/loader.gif"

const LoaderGif: FunctionComponent<ImgHTMLAttributes<HTMLImageElement>> = ({
  alt,
  ...rest
}) => <Image alt={alt} src={Gif} {...rest} />

export default LoaderGif
