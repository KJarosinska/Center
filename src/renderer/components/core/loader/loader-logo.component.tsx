import * as React from "react"
import FunctionComponent from "Renderer/types/function-component.interface"
import { ImgHTMLAttributes } from "react"
import Image from "Renderer/components/core/image/image.component"
import Gif from "Renderer/images/loader.gif"

const LoaderLogo: FunctionComponent<ImgHTMLAttributes<HTMLImageElement>> = ({
  ...rest
}) => <Image data-testid="loader-logo" src={Gif} {...rest} />

export default LoaderLogo
