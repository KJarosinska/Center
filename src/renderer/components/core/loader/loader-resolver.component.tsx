import * as React from "react"
import FunctionComponent from "Renderer/types/function-component.interface"
import {
  LoaderResolverProp,
  LoaderType,
} from "Renderer/components/core/loader/loader.interface"
import Loader from "Renderer/components/core/loader/loader.component"
import LoaderGif from "Renderer/components/core/loader/loader-gif.component"

const LoaderResolver: FunctionComponent<LoaderResolverProp> = ({
  type,
  loaderProps = {},
}) => {
  const { size = 2, ...rest } = loaderProps
  return (
    <>
      {type === LoaderType.Gif ? (
        <LoaderGif {...rest} />
      ) : (
        <Loader size={size} />
      )}
    </>
  )
}

export default LoaderResolver
