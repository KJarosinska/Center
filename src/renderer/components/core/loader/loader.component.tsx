import * as React from "react"
import FunctionComponent from "Renderer/types/function-component.interface"
import {
  LoaderProps,
  LoaderType,
} from "Renderer/components/core/loader/loader.interface"
import LoaderSpinner from "Renderer/components/core/loader/loader-spinner.component"
import LoaderLogo from "Renderer/components/core/loader/loader-logo.component"

const Loader: FunctionComponent<LoaderProps> = ({ type, loaderProps = {} }) => {
  const { size, ...rest } = loaderProps
  return (
    <>
      {type === LoaderType.Gif ? (
        <LoaderLogo {...rest} />
      ) : (
        <LoaderSpinner size={size} />
      )}
    </>
  )
}

export default Loader
