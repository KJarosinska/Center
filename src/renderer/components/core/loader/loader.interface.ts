import { ImgHTMLAttributes } from "react"

export interface LoaderProps {
  size?: number
}

export enum LoaderType {
  Gif,
  Css,
}

export interface LoaderResolverProp {
  type: LoaderType
  loaderProps?: LoaderProps & ImgHTMLAttributes<HTMLImageElement>
}
