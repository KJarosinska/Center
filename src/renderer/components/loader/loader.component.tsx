import * as React from "react"
import FunctionComponent from "Renderer/types/function-component.interface"
import styled, { css } from "styled-components"

const chaseAnimation = css`
  @keyframes chase {
    100% {
      transform: rotate(360deg);
    }
  }
  animation: chase 2.5s infinite linear both;
`

const dotAnimation = css`
  @keyframes chase-dot {
    80%,
    100% {
      transform: rotate(360deg);
    }
  }
  animation: chase-dot 2s infinite ease-in-out both;
`

const dotBeforeAnimation = css`
  @keyframes dot-before {
    50% {
      transform: scale(0.4);
    }
    100%,
    0% {
      transform: scale(1);
    }
  }
  animation: dot-before 2s infinite ease-in-out both;
`

const LoaderWrapper = styled.div<{ size: Size }>`
  width: ${({ size }) => size.width}px;
  height: ${({ size }) => size.height}px;
  position: relative;
  ${chaseAnimation}
`

const LoaderDot = styled.div<{ color: string }>`
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  ${dotAnimation}
  &:before {
    content: "";
    display: block;
    width: 25%;
    height: 25%;
    background-color: ${({ color }) => color};
    border-radius: 100%;
    ${dotBeforeAnimation}
  }
  &:nth-child(1) {
    animation-delay: -1.1s;
  }
  &:nth-child(2) {
    animation-delay: -1s;
  }
  &:nth-child(3) {
    animation-delay: -0.9s;
  }
  &:nth-child(4) {
    animation-delay: -0.8s;
  }
  &:nth-child(5) {
    animation-delay: -0.7s;
  }
  &:nth-child(6) {
    animation-delay: -0.6s;
  }
  &:nth-child(1):before {
    animation-delay: -1.1s;
  }
  &:nth-child(2):before {
    animation-delay: -1s;
  }
  &:nth-child(3):before {
    animation-delay: -0.9s;
  }
  &:nth-child(4):before {
    animation-delay: -0.8s;
  }
  &:nth-child(5):before {
    animation-delay: -0.7s;
  }
  &:nth-child(6):before {
    animation-delay: -0.6s;
  }
`

interface Size {
  height: number
  width: number
}

interface Props {
  size: Size
  loaderColor: string
  numberOfDots?: number
}

const Loader: FunctionComponent<Props> = ({
  size,
  loaderColor,
  numberOfDots = 6,
}) => {
  const arrayOfDots = Array(numberOfDots)
    .fill(0)
    .map((_, index) => (
      <LoaderDot data-testid="dot" color={loaderColor} key={index} />
    ))
  return <LoaderWrapper size={size}>{arrayOfDots}</LoaderWrapper>
}

export default Loader
