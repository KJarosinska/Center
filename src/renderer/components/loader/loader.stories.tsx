import { object, text, withKnobs } from "@storybook/addon-knobs"
import { storiesOf } from "@storybook/react"
import * as React from "react"
import Loader from "Renderer/components/loader/loader.component"
import styled from "styled-components"

const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 500px;
  width: 100%;
`

storiesOf("Components|Loader ", module)
  .addDecorator(withKnobs)
  .add("Checked", () => {
    const size = object("Loader Size Data", {
      height: 40,
      width: 40,
    })
    const loaderColor = text("Loader color", "pink")
    return (
      <Container>
        <Loader size={size} loaderColor={loaderColor} />
      </Container>
    )
  })
