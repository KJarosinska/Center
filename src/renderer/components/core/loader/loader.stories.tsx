import { storiesOf } from "@storybook/react"
import * as React from "react"
import Loader from "Renderer/components/core/loader/loader.component"
import styled from "styled-components"
import LoaderGif from "Renderer/components/core/loader/loader-gif.component"
import LoaderResolver from "Renderer/components/core/loader/loader-resolver.component"
import { LoaderType } from "Renderer/components/core/loader/loader.interface"

const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 50rem;
  width: 100%;
`

const HotLoader = styled(Loader)`
  color: hotpink;
`

storiesOf("Components|Loader ", module)
  .add("Default", () => {
    return (
      <Container>
        <Loader />
      </Container>
    )
  })
  .add("Custom", () => {
    return (
      <Container>
        <HotLoader />
      </Container>
    )
  })
  .add("Loader gif", () => {
    return (
      <Container>
        <LoaderGif height={200} width={200} />
      </Container>
    )
  })
  .add("Loader resolver", () => {
    return (
      <Container>
        <LoaderResolver
          type={LoaderType.Gif}
          loaderProps={{
            height: 100,
            size: 111,
          }}
        />
      </Container>
    )
  })
