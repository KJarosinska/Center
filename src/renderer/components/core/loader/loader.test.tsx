import "@testing-library/jest-dom"
import React from "react"
import { renderWithThemeAndIntl } from "Renderer/utils/render-with-theme-and-intl"
import Loader from "Renderer/components/core/loader/loader.component"
import { LoaderType } from "Renderer/components/core/loader/loader.interface"

test("returns css loader", () => {
  const testId = "loader-css"
  const { getByTestId } = renderWithThemeAndIntl(
    <Loader type={LoaderType.Css} />
  )
  expect(getByTestId(testId)).toBeInTheDocument()
})

test("returns gif loader", () => {
  const testId = "loader-gif"
  const { getByTestId } = renderWithThemeAndIntl(
    <Loader type={LoaderType.Gif} />
  )
  expect(getByTestId(testId)).toBeInTheDocument()
})
