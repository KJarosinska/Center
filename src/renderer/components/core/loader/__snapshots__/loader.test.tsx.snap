// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="sc-AxjAm dclKcW"
  data-testid="loader-css"
  size="4"
>
  <div
    class="sc-AxirZ jaiFcm"
    data-testid="dot"
  />
  <div
    class="sc-AxirZ jaiFcm"
    data-testid="dot"
  />
  <div
    class="sc-AxirZ jaiFcm"
    data-testid="dot"
  />
  <div
    class="sc-AxirZ jaiFcm"
    data-testid="dot"
  />
  <div
    class="sc-AxirZ jaiFcm"
    data-testid="dot"
  />
  <div
    class="sc-AxirZ jaiFcm"
    data-testid="dot"
  />
</div>
`;
